### PR TITLE
Reference range component

### PR DESCRIPTION
**Pull request by rahul1**

### CODE REVIEW

File: packages/core/src/utils.ts [hunk 1]
```diff
@@ -714,3 +715,55 @@ function toPreciseInteger(a: number, precision?: number): number {
   }
   return Math.round(a * Math.pow(10, precision));
 }
+
+/**
+ * Returns a human-readable string for a FHIR Range datatype, taking into account comparators and one-sided ranges
+ * @param range A FHIR Range element
+ * @returns A human-readable string representation of the Range
+ */
+export function formatRangeString(range: Range | undefined): string {
```
Comment (codyebberson) [inline packages/core/src/utils.ts:724]: This has high duplication with code in `RangeDisplay`.  Should be able to squash into a single function: https://github.com/medplum/medplum/blob/main/packages/react/src/RangeDisplay/RangeDisplay.tsx

File: packages/core/src/utils.ts [hunk 2]
```diff
@@ -714,3 +715,55 @@ function toPreciseInteger(a: number, precision?: number): number {
...
+
+  return `${formatQuantityString(range.low)} - ${formatQuantityString(range.high)}`;
+}
+
+/**
+ * Returns a human-readable string for a FHIR Quantity datatype, taking into account units and comparators
+ * @param quantity A FHIR Quantity element
+ * @returns A human-readable string representation of the Quantity
+ */
+export function formatQuantityString(quantity: Quantity | undefined): string {
```
Comment (codyebberson) [inline packages/core/src/utils.ts:745]: Reconcile this with `QuantityDisplay`: https://github.com/medplum/medplum/blob/main/packages/react/src/QuantityDisplay/QuantityDisplay.tsx

**Review by codyebberson — CHANGES_REQUESTED** [review]

File: packages/core/src/utils.ts (hunk 1 above)
Comment (rahul1) [reply packages/core/src/utils.ts:724]: Ok, I think this is a merege issue. I thought I had _moved_ the function, but somehow I duplicated. Will resolve

File: packages/core/src/utils.ts (hunk 2 above)
Comment (rahul1) [reply packages/core/src/utils.ts:745]: Same as above. Thought I had _moved_ the code, but think it got duplicated in the merge. Will resolve and test

File: packages/react/src/ReferenceRangeEditor/ReferenceRangeEditor.tsx [hunk 3]
```diff
@@ -0,0 +1,439 @@
...
+        onClick={(e: React.MouseEvent) => {
+          killEvent(e);
+          addGroup({ id: `group-id-${groupId}`, filters: {} as IntervalGroup['filters'], intervals: [] });
+          setGroupId((id) => id + 1);
+        }}
+      >
+        <IconCirclePlus />
+      </ActionIcon>
+
+      <Button type="submit">Save</Button>
```
Comment (codyebberson) [inline packages/react/src/ReferenceRangeEditor/ReferenceRangeEditor.tsx:94]: For "end of form" buttons, I've been trying to use:

```tsx
<Group position="right">
 <Button type="submit">Save</Button>
</Group>
```

or, if you need more vertical spacing:

```tsx
<Group position="right" mt="xl">
 <Button type="submit">Save</Button>
</Group>
```

File: packages/core/src/utils.ts [hunk 4]
```diff
@@ -638,12 +638,13 @@ export function matchesRange(value: number, range: Range, precision?: number): b
...
 /**
  * Returns the input number rounded to the specified number of digits.
+ * Negative precision indicates rounding to whole numbers. For example preciseRound(123, -1) === 120
  * @param a The input number.
  * @param precision The precision in number of digits.
  * @returns The number rounded to the specified number of digits.
  */
 export function preciseRound(a: number, precision: number): number {
-  return parseFloat(a.toFixed(precision));
+  return toPreciseInteger(a, precision) / Math.pow(10, precision);
```
Comment (codyebberson) [inline packages/core/src/utils.ts:647]: As discussed offline, let's revert this.  `Observation.decimalPrecision` is positive integers only.

**Review by codyebberson — CHANGES_REQUESTED** [review]
Overall, this looks great.  Nice job.

1. Let's revert the negative digits of precision
2. Take a look at using `<Group position="right">` for the "Save" button

**Review by codyebberson — APPROVED** [review]

**Pull request merged**